File: mobile/lib/models/injury_suspension.dart
class InjurySuspension {
  final int? id;
  final int playerId;
  final String playerName;
  final String type;
  final String severity;
  final String description;
  final DateTime startDate;
  final DateTime? estimatedEndDate;
  final String status;
  final String restrictions;

  InjurySuspension({
    this.id,
    required this.playerId,
    required this.playerName,
    required this.type,
    required this.severity,
    required this.description,
    required this.startDate,
    this.estimatedEndDate,
    required this.status,
    required this.restrictions,
  });

  factory InjurySuspension.fromJson(Map<String, dynamic> json) {
    return InjurySuspension(
      id: json['id'],
      playerId: json['playerId'] ?? 0,
      playerName: json['playerName'] ?? '',
      type: json['type'] ?? 'INJURY',
      severity: json['severity'] ?? 'MEDIUM',
      description: json['description'] ?? '',
      startDate: json['startDate'] != null
          ? DateTime.parse(json['startDate'])
          : DateTime.now(),
      estimatedEndDate: json['estimatedEndDate'] != null
          ? DateTime.parse(json['estimatedEndDate'])
          : null,
      status: json['status'] ?? 'ACTIVE',
      restrictions: json['restrictions'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'playerId': playerId,
      'playerName': playerName,
      'type': type,
      'severity': severity,
      'description': description,
      'startDate': startDate.toIso8601String().split('T')[0],
      'estimatedEndDate': estimatedEndDate?.toIso8601String().split('T')[0],
      'status': status,
      'restrictions': restrictions,
    };
  }
}

File: backend/src/main/java/com/club/model/InjurySuspension.java
package com.club.model;

import jakarta.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "injury_suspensions")
public class InjurySuspension {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "player_id", nullable = false)
    private Joueur player;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private AlertType type;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Severity severity;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "start_date", nullable = false)
    private LocalDate startDate;

    @Column(name = "estimated_end_date")
    private LocalDate estimatedEndDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status = Status.ACTIVE;

    @Column(columnDefinition = "TEXT")
    private String restrictions;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    public InjurySuspension() {
    }

    public InjurySuspension(Joueur player, AlertType type, Severity severity, String description, LocalDate startDate,
            LocalDate estimatedEndDate, Status status, String restrictions) {
        this.player = player;
        this.type = type;
        this.severity = severity;
        this.description = description;
        this.startDate = startDate;
        this.estimatedEndDate = estimatedEndDate;
        this.status = status;
        this.restrictions = restrictions;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Joueur getPlayer() {
        return player;
    }

    public void setPlayer(Joueur player) {
        this.player = player;
    }

    public AlertType getType() {
        return type;
    }

    public void setType(AlertType type) {
        this.type = type;
    }

    public Severity getSeverity() {
        return severity;
    }

    public void setSeverity(Severity severity) {
        this.severity = severity;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public LocalDate getStartDate() {
        return startDate;
    }

    public void setStartDate(LocalDate startDate) {
        this.startDate = startDate;
    }

    public LocalDate getEstimatedEndDate() {
        return estimatedEndDate;
    }

    public void setEstimatedEndDate(LocalDate estimatedEndDate) {
        this.estimatedEndDate = estimatedEndDate;
    }

    public Status getStatus() {
        return status;
    }

    public void setStatus(Status status) {
        this.status = status;
    }

    public String getRestrictions() {
        return restrictions;
    }

    public void setRestrictions(String restrictions) {
        this.restrictions = restrictions;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    public enum AlertType {
        INJURY,
        SUSPENSION
    }

    public enum Severity {
        LOW,
        MEDIUM,
        HIGH
    }

    public enum Status {
        ACTIVE,
        RECOVERED
    }
}
